torch/csrc/jit/codegen/cuda/runtime/fused_reduction.cu

```diff
@@ -517,6 +517,19 @@ class ParallelReduce {
       int64_t& count,
       Funcs... funcs);
 
+  // User-visible entry point of grouped grid welford +
+  // broadcast. Mostly the same as reduceGroup, and it would be
+  // possible to combine this to reduceGroup, but it might make the
+  // templated data structures even more complicated and difficult to
+  // understand. For now, keep it as a separate function.
+  //
+  // Unlike reduceGroup, though, the data types of welford ops must be
+  // the same. For example, reduceGroup can be used to reduce half and
+  // float values by passing a tuple of, e.g., LocalTuple<half,
+  // float>, but that's not supported here for implementation
```

Comment:
I don't remember if this is checked in the lowering.

Reply:
Grouping of welford reduction is only supported through `ParallelType::Group` applied to a ForLoop, so it's always the same type. 

In the case of normal reductions, multiple `reductionOp` exprs can be grouped (i.e., horizontal grouping), and they can be of different types.